arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,228 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class EditFeaturesWithFeatureLinkedAnnotationViewController: UIViewController, AGSGeoViewTouchDelegate {
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            // Set the touch delegate.
+            mapView.touchDelegate = self
+            // Create the map with a light gray canvas basemap centered on Loudoun, Virginia.
+            let map = AGSMap(basemapType: .lightGrayCanvasVector, latitude: 39.0204, longitude: -77.4159, levelOfDetail: 18)
+            mapView.map = map
+            loadGeodatabase()
+        }
+    }
+    private var selectedFeature: AGSFeature?
+    private var selectedFeatureIsPolyline = false
+    
+    func loadGeodatabase() {
+        // Load geodatabase from shared resources.
+        let geodatabaseURL = Bundle.main.url(forResource: "loudoun_anno", withExtension: ".geodatabase")!
+        let geodatabase = AGSGeodatabase(fileURL: geodatabaseURL)
+        geodatabase.load { (error: Error?) in
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                guard let map = self.mapView.map else { return }
+                // Create feature layers from tables in the geodatabase.
+                let parcelLinesFeatureLayer = AGSFeatureLayer(featureTable: geodatabase.geodatabaseFeatureTable(withName: "ParcelLines_1")!)
+                let addressPointFeatureLayer = AGSFeatureLayer(featureTable: geodatabase.geodatabaseFeatureTable(withName: "Loudoun_Address_Points_1")!)
+                // Create annotation layers from tables in the geodatabase.
+                let parcelLinesAnnotationLayer = AGSAnnotationLayer(featureTable: geodatabase.geodatabaseAnnotationTable(withTableName: "ParcelLinesAnno_1")!)
+                let addressPointsAnnotationLayer = AGSAnnotationLayer(featureTable: geodatabase.geodatabaseAnnotationTable(withTableName: "Loudoun_Address_PointsAnno_1")!)
+                // Add the feature layers to the map.
+                map.operationalLayers.add(parcelLinesFeatureLayer)
+                map.operationalLayers.add(addressPointFeatureLayer)
+                // Add the annotation layers to the map.
+                map.operationalLayers.add(parcelLinesAnnotationLayer)
+                map.operationalLayers.add(addressPointsAnnotationLayer)
+            }
+        }
+    }
+    
+    // MARK: - AGSGeoViewTouchDelegate
+    // Select the nearest feature or move the point or polyline vertex to the given screen point.
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // If a feature hasn't been selected.
+        if selectedFeature == nil {
+            selectFeature(screenPoint: screenPoint)
+        } else {
+            // Move the feature.
+            if selectedFeatureIsPolyline {
```

Comment:
`selectedFeatureIsPolyline` is a stored property, but it is only read here. Rather than setting it in multiple places, the value is easily derived from the selected feature itself, so it would be better to derive it here, like so:

```swift
if let selectedFeature = selectedFeature {
    if selectedFeature.geometry?.geometryType == .polyline {
        ...
    } else {
        ...
    }
} else {
    ...
}
```

Then `selectedFeatureIsPolyline` wouldn't even be needed and could just be removed.